app/src/androidTest/java/com/github/sdpcoachme/messaging/ChatActivityTest.kt

```diff
@@ -0,0 +1,315 @@
+import com.github.sdpcoachme.messaging.ChatActivity.TestTags.Companion.CHAT_BOX
+import com.github.sdpcoachme.messaging.ChatActivity.TestTags.Companion.CHAT_FIELD
+import com.github.sdpcoachme.messaging.ChatActivity.TestTags.Companion.CHAT_MESSAGE
+import com.github.sdpcoachme.messaging.ChatActivity.TestTags.Companion.CONTACT_FIELD
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.lessThan
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.Duration
+import java.time.LocalDateTime
+
+@RunWith(AndroidJUnit4::class)
+class ChatActivityTest {
+
+    @get:Rule
+    val composeTestRule = createEmptyComposeRule()
+
+    private val toUser = UserInfo(
+        "Jane",
+        "Doe",
+        "to@email.com",
+        "0987654321",
+        LAUSANNE,
+        true,
+        emptyList(),
+        emptyList()
+    )
+    private lateinit var database: Database
+    private val currentUser = UserInfo(
+        "John",
+        "Doe",
+        "example@email.com",
+        "0123456789",
+        NEW_YORK,
+        false,
+        emptyList(),
+        emptyList()
+    )
+    private val chatId = (currentUser.email + toUser.email)
+
+
+    @Before
+    fun setup() {
+        database = (InstrumentationRegistry.getInstrumentation().targetContext.applicationContext as CoachMeApplication).database
+        database.setCurrentEmail(currentUser.email)
+        database.updateUser(toUser)
+    }
+
+    @After
+    fun tearDown() {
+        if (database is MockDatabase) {
+            (database as MockDatabase).restoreDefaultChatSetup()
+            println("MockDatabase was torn down")
+        }
+    }
+
+    @Test
+    fun startingElementsArePresent() {
+        val chatIntent = Intent(ApplicationProvider.getApplicationContext(), ChatActivity::class.java)
+        chatIntent.putExtra("toUserEmail", toUser.email)
+
+        ActivityScenario.launch<ChatActivity>(chatIntent).use {
+            composeTestRule.onNodeWithTag(BACK).assertIsDisplayed()
+            composeTestRule.onNodeWithTag(CONTACT_FIELD.LABEL, useUnmergedTree = true).assertTextEquals(toUser.firstName + " " + toUser.lastName)
+            composeTestRule.onNodeWithTag(CHAT_FIELD.LABEL, useUnmergedTree = true).assertIsDisplayed()
+            composeTestRule.onNodeWithTag(CHAT_BOX.CONTAINER).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun whenScrolledToTheBottomScrollButtonIsNotDisplayed() {
+        val chatIntent = Intent(ApplicationProvider.getApplicationContext(), ChatActivity::class.java)
+        chatIntent.putExtra("toUserEmail", toUser.email)
+
+        ActivityScenario.launch<ChatActivity>(chatIntent).use {
+            composeTestRule.onNodeWithTag(SCROLL_TO_BOTTOM, useUnmergedTree = true).assertDoesNotExist()
```

Comment:
Maybe, there exists a possibility to perform a scroll down first before doing this check, just to be sure that it is actually at the bottom. I assume it always starts at the bottom, but this way the test would be a tiny bit more convincing.

Reply:
Thank you for this comment. As the chat's scroll starts all the way at the bottom of the chat, the scroll-button should not be displayed when launching this activity. I will add a comment to the test to clarify it.